numba/core/boxing.py

```diff
@@ -1105,3 +1107,186 @@ def unbox_typeref(typ, val, c):
+            handle_failure()
+
+        ct_cast = object_getattr_safely(ctypes_module, 'cast')
+        with early_exit_if_null(c.builder, stack, ct_cast):
+            handle_failure()
+
+        ct_voidptr_ty = object_getattr_safely(ctypes_module, 'c_void_p')
+        with early_exit_if_null(c.builder, stack, ct_voidptr_ty):
+            handle_failure()
+
+        # This wires in the fnptrs refered to by name
+        def wire_in_fnptrs(name):
+            # Find the CFunctionType function
+            interface_next_fn = c.pyapi.object_getattr_string(
+                ctypes_binding, name)
+
+            extra_refs.append(interface_next_fn)
+            with early_exit_if_null(c.builder, stack, interface_next_fn):
+                handle_failure()
+
+            # Want to do ctypes.cast(CFunctionType, ctypes.c_void_p), create an
+            # args tuple for that.
+            extra_refs.append(ct_voidptr_ty)
+            args = c.pyapi.tuple_pack([interface_next_fn, ct_voidptr_ty])
+            with early_exit_if_null(c.builder, stack, args):
+                handle_failure()
+            extra_refs.append(ct_voidptr_ty)
+
+            # Call ctypes.cast()
+            interface_next_fn_casted = c.pyapi.call(ct_cast, args)
+
+            # Fetch the .value attr on the resulting ctypes.c_void_p for storage
+            # in the function pointer slot.
+            interface_next_fn_casted_value = object_getattr_safely(
+                interface_next_fn_casted, 'value')
+            with early_exit_if_null(c.builder, stack, interface_next_fn_casted_value):
+                handle_failure()
+
+            # Wire up
+            setattr(struct_ptr, f'fnptr_{name}',
+                    c.unbox(types.uintp, interface_next_fn_casted_value).value)
+
+
+        wire_in_fnptrs('next_double')
+        wire_in_fnptrs('next_uint64')
+        wire_in_fnptrs('next_uint32')
+
+        clear_extra_refs()
+
+    return NativeValue(struct_ptr._getvalue(), is_error=c.builder.load(is_error_ptr))
+
+_bit_gen_type = types.NumPyRandomBitGeneratorType('bit_generator')
+
+@unbox(types.NumPyRandomGeneratorType)
+def unbox_numpy_random_generator(typ, obj, c):
+    """
+    Here we're creating a NumPyRandomGeneratorType StructModel with following fields:
+    * ('bit_generator', _bit_gen_type): The unboxed BitGenerator associated with
+                                        this Generator object instance.
+    * ('parent', types.pyobject): Pointer to the original Generator PyObject.
+    * ('meminfo', types.MemInfoPointer(types.voidptr)): The information about the memory
+        stored at the pointer (to the original Generator PyObject). This is useful for
+        keeping track of reference counts within the Python runtime. Helps prevent cases
+        where deletion happens in Python runtime without NRT being awareness of it. 
+    """
+    is_error_ptr = cgutils.alloca_once_value(c.builder, cgutils.false_bit)
+    fail_obj = c.context.get_constant_null(typ)
```

Comment:
unused
```suggestion
```

Reply:
Was removed in [40e87e9](https://github.com/numba/numba/pull/8031/commits/40e87e903789460997195bb3d639ee48213fd13b)